### PR TITLE
Bug 1184164 - adding marionette integration testing for gaia-switch

### PR DESCRIPTION
**Pull request by yzen**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,4 +21,5 @@ env:
   matrix:
     - CI_ACTION="npm run lint"
     - CI_ACTION="npm test"
+    - CI_ACTION="npm run-script test-integration"
```
Comment (wilsonpage) [inline .travis.yml]: `CI_ACTION="npm run test-integration"`

Comment (wilsonpage) [inline .travis.yml]: For clarity we should break the test into categories:

- `test-lint`
- `test-unit`
- `test-integration`

File: test/test_integration.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
```
Comment (wilsonpage) [inline test/test_integration.html]: Can we use `test-integration.html` to align with above naming and existing gaia-components file naming convention?

File: test/test_integration.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/* global marionette, setup, test */
```
Comment (wilsonpage) [inline test/test_integration.js]: Same as above

File: README.md [hunk 4]
```diff
@@ -47,6 +47,10 @@ If your would like tests to run on file change use:
 
 `$ npm run test-dev`
 
+If your would like run integration tests use:
```
Comment (gasolin) [inline README.md]: tests, use

File: README.md [hunk 5]
```diff
@@ -47,6 +47,10 @@ If your would like tests to run on file change use:
...
+
+`$ npm run-script test-integration`
```
Comment (gasolin) [inline README.md]: `$ npm run test-integration`

File: package.json [hunk 6]
```diff
@@ -13,12 +13,17 @@
...
+    "mozilla-download": "^1.0.5",
+    "marionette-firefox-host": "1.0.3",
+    "marionette-helper": "0.3.1",
+    "marionette-js-runner": "1.1.2"
   },
   "scripts": {
     "install": "bower install",
     "lint": "./node_modules/jshint/bin/jshint gaia-switch.js test/*.js",
     "test": "./node_modules/karma/bin/karma start test/karma.conf.js --single-run",
+    "test-integration": "./node_modules/.bin/marionette-mocha --host marionette-firefox-host --runtime /Applications/FirefoxNightly.app/Contents/MacOS/firefox-bin --timeout 6000s test/test_integration.js",
```
Comment (gasolin) [inline package.json]: It's too tight with Mac, maybe we can reuse karma-firefox-launcher?

Comment (wilsonpage) [inline package.json]: 'too tight'?

File: .travis.yml (hunk 1 above)
Comment (wilsonpage) [inline .travis.yml]: Any update on this?

File: README.md [hunk 7]
```diff
@@ -41,14 +41,18 @@ Disabled
...
-3. `$ npm test`
+3. `$ npm run test-unit`
 
 If your would like tests to run on file change use:
 
-`$ npm run test-dev`
+`$ npm run test-unit-dev`
+
+If your would like run integration tests, use:
+`$ export FIREFOX_NIGHTLY_BIN=/absolute/path/to/nightly/firefox-bin`
```
Comment (wilsonpage) [inline README.md]: Would be nice if we could automate this, but I won't block on that :)

File: test/test-integration.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        'browser.feeds.showFirstRunUI': false,
+        'browser.shell.checkDefaultBrowser': false,
+        'browser.uitour.enabled': false,
+        'devtools.chrome.enabled': true,
+        'devtools.debugger.remote-enabled': true,
+
+        'startup.homepage_welcome_url': __dirname + '/test-integration.html',
+
+        'dom.webcomponents.enabled': true,
+        'dom.w3c_touch_events.enabled': 1
```
Comment (wilsonpage) [inline test/test-integration.js]: Can you comment each one of these prefs to tell us what they are for.

File: test/test-integration.js [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+      }
+    },
+    desiredCapabilities: {
+      raisesAccessibilityExceptions: true
+    }
+  });
```
Comment (wilsonpage) [inline test/test-integration.js]: NIT: Insert empty line after here

**Pull request merged**